app/models/ability.rb

```diff
@@ -50,7 +50,12 @@ def not_signed_in
         registration.new_record?
       end
 
-      can [:new, :create], Registration do |registration|
+      can [:new], Registration do |registration|
+        conf = registration.conference
+        conf.registration_open? && registration.new_record? && !conf.registration_limit_exceeded?
+      end
+
+      can [:create], Registration do |registration|
```

Comment:
ability.rb should still fully reflect what is happening, ie if the user has access to a specific action, or not.

And please change variable `conf` into `conference`.